docs/DeveloperGuide.md

```diff
@@ -2,14 +2,372 @@
+    private static final String[] ADD_ARGUMENTS = {"courseCode", "semester"};
+
+    public AddCommandMetadata() {
+        super(ADD_KEYWORD, ADD_ARGUMENTS);
+    }
+
+    // Add Command Creator
+    @Override
+    protected Command createCommandInstance(Map<String, String> args) {
+        try {
+            String moduleCode = args.getOrDefault("courseCode", "COURSECODE_ERROR");
+            String semester = args.getOrDefault("semester", "SEMESTER_ERROR");
+            int semesterInt = Integer.parseInt(semester);
+
+            return new AddCommand(moduleCode, semesterInt);
+        } catch (ModuleNotFoundException e) {
+            LOGGER.log(Level.SEVERE, "An error occurred: " + e.getMessage());
+            System.out.println("An error occurred: " + e.getMessage());
+        }
+        return new InvalidCommand();
+    }
+}
+  ```
+Design & Implementation
+Storage Class
+
+The Storage class is crucial for persisting user data between sessions, ensuring that module information and user preferences are not lost even after the application is closed. It interacts with the file system to load and save data, employing error handling to manage potential IO exceptions gracefully.
+Key Responsibilities
+
+- **File Management:** Ensures necessary directories and files exist at application startup or creates them if they don't.
+- **Data Persistence:** Serializes and deserializes user data, including modules and user information, to and from a designated file.
+- **Error Handling:** Captures and throws custom exceptions to signal file access or data integrity issues, facilitating robust error management in higher-level components.
+
+**Implementation Details**
+
+- **Initialization:** Verifies the presence of required directories/files or creates them. This is crucial for first-time application runs on a user's machine.
+
+- **Saving Data:**
+    The `saveModulesToFile` method serializes the current state of moduleList and user into a human-readable format (or a structured format like JSON/XML, depending on implementation) and writes it to a file. This method is invoked at critical points, such as application shutdown or after any operation that alters user data.
+
+- **Loading Data:**
+    The `loadDataFromFile` method reads the file contents, deserializes them back into the application's data structures (moduleList and user), and ensures that the application state reflects the persisted data. This method is typically called at application startup.
+
+- **Data Integrity and Error Management:**
+    Implements `try-catch` blocks to manage `IOExceptions` and custom exceptions, ensuring the application can handle and recover from unexpected issues during file operations.
+
+   ### UML Diagram
+
+   A simplified UML diagram for the Storage class and its interaction with the Module, User, and exception classes is shown below:
+-    ![View Storage Class](diagrams/Storage.png)
+
+   ### Integration with FAP
```

Comment:
You can consider editing this part of the developer guide as the "###" used here is not recognised as the "header"